{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## About\n",
    "\n",
    "In lights of the success of the ProTargetMiner study (Saei et al. - ProTargetMiner as a proteome signature library of anticancer molecules for functional discovery) further data mining from the same data set, but extended with the proteome of dying cells will be conducted in this project. The overall goal of this project is to analyze the proteome for living and dying cells to understand cell death, specifically to find proteome signatures relating to cell death regardless of which treatment or cell line the cells belongs to, as well as be able to identify different apoptotic processes in the cell. The dataset consists of living and dead cells of three cell lines (cancer cell lines; A549, RKO and MCF-7). The apoptotic process has been introduced by nine different cancer treatments (8-zaguanine, Raltitrexed, Topotecan, Floxuridine, Nutlin, Dasatinib, Gefitinib, Vincristine and Bortezomib), and there are three replicates, meaning that in total there are 60 different conditions in 3 replicates.  \n",
    "\n",
    "For this project there is a couple of things we would like to investigate:\n",
    "\n",
    "1) The number of apoptotic pathways. It has been suggested that there are at least two major signalling pathways trigger apoptotic cell death in [Geen et al.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4665079/), but we would like to investigate exactly how many there are. \n",
    "2) Proteome signature for surviving and dying cells regardless of treatment or cell line. To do this we would need to investigate target regulation in dying vs. surviving cells for each treatment and cell line as well.   \n",
    "\n",
    "## Goals\n",
    "- Count amount of up- and downregulated proteins/peptides for life and death states.\n",
    "- Check if there are proteins which are always up- or down-regulated for regardless of drug treatment. \n",
    "- Check the correlations between protein abundances. \n",
    "- Measure variance and co-variance of proteins. \n",
    "\n",
    "\n",
    "\n",
    "The project is related to:\n",
    "\n",
    "[ProTargetMiner as a proteome signature library of anticancer molecules for functional discovery](https://www.nature.com/articles/s41467-019-13582-8)\n",
    "\n",
    "[Comparative Proteomics of Dying and Surviving Cancer Cells Improves the Identification of Drug Targets and Sheds Light on Cell Life / Death Decisions](https://pubmed.ncbi.nlm.nih.gov/29572246/)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Problem\n",
    "\n",
    "The apoptotic process is related to many human diseases, which may result when cells die that shouldn't and other live that should die. Modulation of apoptotic processes may therefore offer valuable methods of treatment ([Renehan et al.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1120576/)). Understanding the apoptotic process is therefore very important. A dataset used in [Saei et al.](https://www.nature.com/articles/s41467-019-13582-8) proved to be of valuable for uncovering protein signatures relating to drug compound targets and actions mechnisms. Therefore and extension of this dataset with  \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2020-10-21 Wednesday\n",
    "### 17:34 Some thoughts when processing proteins with my GO-id function.\n",
    "I am using the Leading_razor_protein as my search protein. The GO search goes bad when we have entries with the following formatting:\n",
    "\n",
    "P51608-2\n",
    "Q9Y5Q8-3\n",
    "CON__Q2KJC7\n",
    "A0A0C4DGS5\n",
    "CON__ENSEMBL:ENSBTAP00000007350\n",
    "D3DTX6\n",
    "\n",
    "But testing P51608-2 --> P51608 I find can get a GO-match. I guess this has to do with the suffix \"-2\"l. I also notice that often when P51608-2 is the Leading_razor_protein; the protein column also contains P51608, P51608-2; P51608-3, etc. I will investigate what the suffixes and errors are later.\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 15:48 Data description\n",
    "Just finished coding a function to infer GO from protein names with uniprot, but there are some searching problems. But first lets describe the data.\n",
    "\n",
    "There are four files:\n",
    "\n",
    "- peptides tryptic.txt\n",
    "- peptides.txt\n",
    "- proteinGroups tryptic.txt\n",
    "- proteinGroups.txt\n",
    "\n",
    "There are two searches performed; one tryptic (I guess the tryptic files are the tryptic searches) and one semi-tryptic. I've been told this is performed because in cell death proteases get active and degrade some proteins. \n",
    "\n",
    "The data code is:\n",
    "\n",
    "\n",
    "Reporter intensity corrected 0 A549_S_Rep1\n",
    "\n",
    "0 is the TMT tag code, it goes from 0-9 and covers a control (0) and the rest are 9 drugs in the following order: \n",
    "\n",
    "Control\n",
    "8-zaguanine\n",
    "Raltitrexed\n",
    "Topotecan\n",
    "Floxuridine\n",
    "Nutlin\n",
    "Dasatinib\n",
    "Gefitinib\n",
    "Vincristine\n",
    "Bortezomib\n",
    "\n",
    "A549, RKO and MCF-7 are the cell lines. \n",
    "\n",
    "S stands for \"Surviving\" meaning attached cells after treatment, and D stands for \"Dying\", those cells that detached from the plate after treatment with the anticancer drugs.\n",
    "Rep1-3 are the replicate numbers, all the experiments are in 3 replicates. \n",
    "\n",
    "The surviving cell data comes from this paper ([ProTargetMiner as a proteome signature library of anticancer molecules for functional discovery](https://www.nature.com/articles/s41467-019-13582-8)). The dying cell data is new. \n",
    "\n",
    "The head of the unprocessed data is shown in the table below:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "data_loc = \"~/git/lifeAndDeath/data/knitr/\"\n",
    "\n",
    "peptides = pd.read_csv(data_loc + \"peptides_knitr.txt\", sep = \"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sequence</th>\n",
       "      <th>N-term cleavage window</th>\n",
       "      <th>C-term cleavage window</th>\n",
       "      <th>Amino acid before</th>\n",
       "      <th>First amino acid</th>\n",
       "      <th>Second amino acid</th>\n",
       "      <th>Second last amino acid</th>\n",
       "      <th>Last amino acid</th>\n",
       "      <th>Amino acid after</th>\n",
       "      <th>A Count</th>\n",
       "      <th>...</th>\n",
       "      <th>Reverse</th>\n",
       "      <th>Potential contaminant</th>\n",
       "      <th>id</th>\n",
       "      <th>Protein group IDs</th>\n",
       "      <th>Mod. peptide IDs</th>\n",
       "      <th>Evidence IDs</th>\n",
       "      <th>MS/MS IDs</th>\n",
       "      <th>Best MS/MS</th>\n",
       "      <th>Oxidation (M) site IDs</th>\n",
       "      <th>MS/MS Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAAAAAAAAA</td>\n",
       "      <td>LVSENAGRAAAAAAAA</td>\n",
       "      <td>AAAAAAAAAAAAAGAG</td>\n",
       "      <td>R</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>10</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>4743;2687;1346;7153;8131</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAAAAAAAAAA</td>\n",
       "      <td>LVSENAGRAAAAAAAA</td>\n",
       "      <td>AAAAAAAAAAAAGAGA</td>\n",
       "      <td>R</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>11</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>4743;2687;1346;8131</td>\n",
       "      <td>1</td>\n",
       "      <td>1;2;3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAAAAAAAAAAAAAAGAGAGAK</td>\n",
       "      <td>LVSENAGRAAAAAAAA</td>\n",
       "      <td>AGAGAGAKQTPADGEA</td>\n",
       "      <td>R</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>K</td>\n",
       "      <td>Q</td>\n",
       "      <td>18</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>4743</td>\n",
       "      <td>2</td>\n",
       "      <td>4;5;6;7;8;9;10;11;12;13;14;15;16;17;18;19;20;2...</td>\n",
       "      <td>2;3;4;5;6;7;8;9;10;11;12;13;14;15;16</td>\n",
       "      <td>13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAAAAAAAAAAAATATAGPR</td>\n",
       "      <td>RLLQGAAAAAAAAAAA</td>\n",
       "      <td>ATATAGPRGEAPPPPP</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>P</td>\n",
       "      <td>R</td>\n",
       "      <td>G</td>\n",
       "      <td>15</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>8875</td>\n",
       "      <td>3</td>\n",
       "      <td>29;30;31;32</td>\n",
       "      <td>17;18;19</td>\n",
       "      <td>19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAAAAAAAAAAK</td>\n",
       "      <td>LFAPASAAAAAAAAAA</td>\n",
       "      <td>AAAAAAAKGALEGAAG</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>K</td>\n",
       "      <td>G</td>\n",
       "      <td>11</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>712;11411</td>\n",
       "      <td>4</td>\n",
       "      <td>33;34</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>AAAAAAAAAAGAAGGR</td>\n",
       "      <td>________________</td>\n",
       "      <td>AAGAAGGRGSGPGRRR</td>\n",
       "      <td>M</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>R</td>\n",
       "      <td>G</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>7153</td>\n",
       "      <td>5</td>\n",
       "      <td>35;36;37;38;39;40;41;42;43;44;45;46;47;48;49;5...</td>\n",
       "      <td>21;22;23;24;25;26;27;28;29;30;31;32;33;34;35;3...</td>\n",
       "      <td>26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>AAAAAAAAAAGAAGGRGSGPGR</td>\n",
       "      <td>________________</td>\n",
       "      <td>GRGSGPGRRRHLVPGA</td>\n",
       "      <td>M</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>R</td>\n",
       "      <td>R</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>7153</td>\n",
       "      <td>6</td>\n",
       "      <td>82;83;84</td>\n",
       "      <td>61;62;63</td>\n",
       "      <td>63</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>AAAAAAAAAAGEGAR</td>\n",
       "      <td>________________</td>\n",
       "      <td>AAAGEGARSPSPAAVS</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>R</td>\n",
       "      <td>S</td>\n",
       "      <td>11</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>7012</td>\n",
       "      <td>7</td>\n",
       "      <td>85</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>AAAAAAAAAGEGAR</td>\n",
       "      <td>________________</td>\n",
       "      <td>AAAGEGARSPSPAAVS</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>R</td>\n",
       "      <td>S</td>\n",
       "      <td>10</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>7012</td>\n",
       "      <td>8</td>\n",
       "      <td>86;87</td>\n",
       "      <td>65</td>\n",
       "      <td>65</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9 rows × 706 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Sequence N-term cleavage window C-term cleavage window  \\\n",
       "0              AAAAAAAAAA       LVSENAGRAAAAAAAA       AAAAAAAAAAAAAGAG   \n",
       "1             AAAAAAAAAAA       LVSENAGRAAAAAAAA       AAAAAAAAAAAAGAGA   \n",
       "2  AAAAAAAAAAAAAAAGAGAGAK       LVSENAGRAAAAAAAA       AGAGAGAKQTPADGEA   \n",
       "3    AAAAAAAAAAAAATATAGPR       RLLQGAAAAAAAAAAA       ATATAGPRGEAPPPPP   \n",
       "4            AAAAAAAAAAAK       LFAPASAAAAAAAAAA       AAAAAAAKGALEGAAG   \n",
       "5        AAAAAAAAAAGAAGGR       ________________       AAGAAGGRGSGPGRRR   \n",
       "6  AAAAAAAAAAGAAGGRGSGPGR       ________________       GRGSGPGRRRHLVPGA   \n",
       "7         AAAAAAAAAAGEGAR       ________________       AAAGEGARSPSPAAVS   \n",
       "8          AAAAAAAAAGEGAR       ________________       AAAGEGARSPSPAAVS   \n",
       "\n",
       "  Amino acid before First amino acid Second amino acid Second last amino acid  \\\n",
       "0                 R                A                 A                      A   \n",
       "1                 R                A                 A                      A   \n",
       "2                 R                A                 A                      A   \n",
       "3                 A                A                 A                      P   \n",
       "4                 A                A                 A                      A   \n",
       "5                 M                A                 A                      G   \n",
       "6                 M                A                 A                      G   \n",
       "7                 A                A                 A                      A   \n",
       "8                 A                A                 A                      A   \n",
       "\n",
       "  Last amino acid Amino acid after  A Count  ...  Reverse  \\\n",
       "0               A                A       10  ...      NaN   \n",
       "1               A                A       11  ...      NaN   \n",
       "2               K                Q       18  ...      NaN   \n",
       "3               R                G       15  ...      NaN   \n",
       "4               K                G       11  ...      NaN   \n",
       "5               R                G       12  ...      NaN   \n",
       "6               R                R       12  ...      NaN   \n",
       "7               R                S       11  ...      NaN   \n",
       "8               R                S       10  ...      NaN   \n",
       "\n",
       "   Potential contaminant  id         Protein group IDs  Mod. peptide IDs  \\\n",
       "0                    NaN   0  4743;2687;1346;7153;8131                 0   \n",
       "1                    NaN   1       4743;2687;1346;8131                 1   \n",
       "2                    NaN   2                      4743                 2   \n",
       "3                    NaN   3                      8875                 3   \n",
       "4                    NaN   4                 712;11411                 4   \n",
       "5                    NaN   5                      7153                 5   \n",
       "6                    NaN   6                      7153                 6   \n",
       "7                    NaN   7                      7012                 7   \n",
       "8                    NaN   8                      7012                 8   \n",
       "\n",
       "                                        Evidence IDs  \\\n",
       "0                                                  0   \n",
       "1                                              1;2;3   \n",
       "2  4;5;6;7;8;9;10;11;12;13;14;15;16;17;18;19;20;2...   \n",
       "3                                        29;30;31;32   \n",
       "4                                              33;34   \n",
       "5  35;36;37;38;39;40;41;42;43;44;45;46;47;48;49;5...   \n",
       "6                                           82;83;84   \n",
       "7                                                 85   \n",
       "8                                              86;87   \n",
       "\n",
       "                                           MS/MS IDs  Best MS/MS  \\\n",
       "0                                                  0           0   \n",
       "1                                                  1           1   \n",
       "2               2;3;4;5;6;7;8;9;10;11;12;13;14;15;16          13   \n",
       "3                                           17;18;19          19   \n",
       "4                                                 20          20   \n",
       "5  21;22;23;24;25;26;27;28;29;30;31;32;33;34;35;3...          26   \n",
       "6                                           61;62;63          63   \n",
       "7                                                 64          64   \n",
       "8                                                 65          65   \n",
       "\n",
       "   Oxidation (M) site IDs  MS/MS Count  \n",
       "0                     NaN            1  \n",
       "1                     NaN            1  \n",
       "2                     NaN           15  \n",
       "3                     NaN            3  \n",
       "4                     NaN            1  \n",
       "5                     NaN           40  \n",
       "6                     NaN            3  \n",
       "7                     NaN            1  \n",
       "8                     NaN            1  \n",
       "\n",
       "[9 rows x 706 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "peptides"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This data is converted to seaborn-melted format, values have been log2FC-transformed where log2FC(A,B) is used, where the A is the untreated sample for each {cell_line}_{state}_{replicate} and the B is the treated for each {cell_line}_{state}_{replicate}. The data is tresholded with following parameters:\n",
    "\n",
    "- PEP < 0.01\n",
    "- Missed_cleaveges < 2\n",
    "- Reporter_intensity_count > 0\n",
    "- Reporter_intensity_corrected not in [-inf, inf],\n",
    "\n",
    "where I suspect -inf and inf is caused by log(0) values. \n",
    "\n",
    "The function \"log2FC_data_peptide\" in /scripts/preprocessor.py (script needs cleanup) is used for log2FC. /scripts/preprocessor.py also has the seaborn-melted format converter code in it (but it is still not in a function).\n",
    "\n",
    "The converted data is shown below:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "melted = pd.read_csv(data_loc + \"melted_treshold_knitr.csv\", sep = \"\\t\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Leading_razor_protein</th>\n",
       "      <th>Proteins</th>\n",
       "      <th>Reporter_intensity_count</th>\n",
       "      <th>Cell_line</th>\n",
       "      <th>Treatment</th>\n",
       "      <th>State</th>\n",
       "      <th>Replicate</th>\n",
       "      <th>Reporter_intensity_corrected</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Q86U42</td>\n",
       "      <td>Q86U42;Q86U42-2</td>\n",
       "      <td>2</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Q9Y4H2</td>\n",
       "      <td>Q9Y4H2</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>O75822</td>\n",
       "      <td>O75822;O75822-2;O75822-3</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>O75822</td>\n",
       "      <td>O75822;O75822-2;O75822-3</td>\n",
       "      <td>3</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P36578</td>\n",
       "      <td>P36578;H3BM89;H3BU31</td>\n",
       "      <td>7</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>P51608-2</td>\n",
       "      <td>P51608-2;B5MCB4</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Q96P70</td>\n",
       "      <td>Q96P70</td>\n",
       "      <td>2</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>P28482</td>\n",
       "      <td>P28482;P28482-2</td>\n",
       "      <td>2</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>P09417</td>\n",
       "      <td>P09417;P09417-2;B7Z415;D6RGG7;D6RHJ7</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Q01167</td>\n",
       "      <td>Q01167;Q01167-2;Q01167-3</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Q6SPF0</td>\n",
       "      <td>Q6SPF0</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Q86X55-1</td>\n",
       "      <td>Q86X55-1;Q86X55;Q86X55-2;K7EPK1;K7EK20</td>\n",
       "      <td>2</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>O76031</td>\n",
       "      <td>O76031</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Q9ULW3</td>\n",
       "      <td>Q9ULW3</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>P23610</td>\n",
       "      <td>P23610</td>\n",
       "      <td>2</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Q8WUQ7</td>\n",
       "      <td>Q8WUQ7;Q8WUQ7-2</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Q7L5D6</td>\n",
       "      <td>Q7L5D6</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>P27361</td>\n",
       "      <td>P27361;E9PQW4;P27361-2;P27361-3;E9PJF0</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Q9Y2U8</td>\n",
       "      <td>Q9Y2U8</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Q9Y2U8</td>\n",
       "      <td>Q9Y2U8</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Q5VYK3</td>\n",
       "      <td>Q5VYK3;R4GMY1</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>J3KPC8</td>\n",
       "      <td>J3KPC8</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Q13049</td>\n",
       "      <td>Q13049;Q5JVY0</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Q9P258</td>\n",
       "      <td>Q9P258</td>\n",
       "      <td>2</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Q9H3U1</td>\n",
       "      <td>Q9H3U1;Q9H3U1-3;Q9H3U1-2</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Q96E14</td>\n",
       "      <td>Q96E14;I3L2E0</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Q99615</td>\n",
       "      <td>Q99615;K7ESP1;K7ELJ8;K7ENB7</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Q99615</td>\n",
       "      <td>Q99615;K7ESP1;K7ELJ8</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Q6QNY1</td>\n",
       "      <td>Q6QNY1</td>\n",
       "      <td>2</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>O43324</td>\n",
       "      <td>O43324;D6RBD7;C9J1V9;O43324-2</td>\n",
       "      <td>3</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Q6NXT1</td>\n",
       "      <td>Q6NXT1;D3YTC9;C9JU48</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Q9HCJ3-2</td>\n",
       "      <td>Q9HCJ3-2;Q9HCJ3</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Q9UMN6</td>\n",
       "      <td>Q9UMN6;Q9UMN6-2</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>Q5SRE5</td>\n",
       "      <td>Q5SRE5;Q5SRE5-2</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>Q9Y2U8</td>\n",
       "      <td>Q9Y2U8</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>Q96IV0</td>\n",
       "      <td>Q96IV0;Q96IV0-2;Q96IV0-3</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>Q15596</td>\n",
       "      <td>Q15596;E7EWM1</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>Q12972</td>\n",
       "      <td>Q12972</td>\n",
       "      <td>2</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>Q96L91-5</td>\n",
       "      <td>Q96L91-5;Q96L91-2;Q96L91;Q96L91-3</td>\n",
       "      <td>2</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>P52701</td>\n",
       "      <td>P52701;P52701-3;P52701-2;F8WAX8;F8W7G9</td>\n",
       "      <td>2</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>P42566</td>\n",
       "      <td>P42566;B1AUU8</td>\n",
       "      <td>2</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>P55036</td>\n",
       "      <td>P55036;H0Y3Y9</td>\n",
       "      <td>2</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>Q5VWC4</td>\n",
       "      <td>Q5VWC4</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>A1X283</td>\n",
       "      <td>A1X283</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>Q8N1B4</td>\n",
       "      <td>Q8N1B4;E9PI03</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>Q9BQE4</td>\n",
       "      <td>Q9BQE4;A0A182DWI4;E9PN30</td>\n",
       "      <td>2</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>Q96K37</td>\n",
       "      <td>Q96K37</td>\n",
       "      <td>2</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>A0A087WUC6</td>\n",
       "      <td>A0A087WUC6;Q15005;E9PL01;E9PI68</td>\n",
       "      <td>1</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>Q8N1G4</td>\n",
       "      <td>Q8N1G4</td>\n",
       "      <td>2</td>\n",
       "      <td>A549</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>Rep1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Leading_razor_protein                                Proteins  \\\n",
       "0                 Q86U42                         Q86U42;Q86U42-2   \n",
       "1                 Q9Y4H2                                  Q9Y4H2   \n",
       "2                 O75822                O75822;O75822-2;O75822-3   \n",
       "3                 O75822                O75822;O75822-2;O75822-3   \n",
       "4                 P36578                    P36578;H3BM89;H3BU31   \n",
       "5               P51608-2                         P51608-2;B5MCB4   \n",
       "6                 Q96P70                                  Q96P70   \n",
       "7                 P28482                         P28482;P28482-2   \n",
       "8                 P09417    P09417;P09417-2;B7Z415;D6RGG7;D6RHJ7   \n",
       "9                 Q01167                Q01167;Q01167-2;Q01167-3   \n",
       "10                Q6SPF0                                  Q6SPF0   \n",
       "11              Q86X55-1  Q86X55-1;Q86X55;Q86X55-2;K7EPK1;K7EK20   \n",
       "12                O76031                                  O76031   \n",
       "13                Q9ULW3                                  Q9ULW3   \n",
       "14                P23610                                  P23610   \n",
       "15                Q8WUQ7                         Q8WUQ7;Q8WUQ7-2   \n",
       "16                Q7L5D6                                  Q7L5D6   \n",
       "17                P27361  P27361;E9PQW4;P27361-2;P27361-3;E9PJF0   \n",
       "18                Q9Y2U8                                  Q9Y2U8   \n",
       "19                Q9Y2U8                                  Q9Y2U8   \n",
       "20                Q5VYK3                           Q5VYK3;R4GMY1   \n",
       "21                J3KPC8                                  J3KPC8   \n",
       "22                Q13049                           Q13049;Q5JVY0   \n",
       "23                Q9P258                                  Q9P258   \n",
       "24                Q9H3U1                Q9H3U1;Q9H3U1-3;Q9H3U1-2   \n",
       "25                Q96E14                           Q96E14;I3L2E0   \n",
       "26                Q99615             Q99615;K7ESP1;K7ELJ8;K7ENB7   \n",
       "27                Q99615                    Q99615;K7ESP1;K7ELJ8   \n",
       "28                Q6QNY1                                  Q6QNY1   \n",
       "29                O43324           O43324;D6RBD7;C9J1V9;O43324-2   \n",
       "30                Q6NXT1                    Q6NXT1;D3YTC9;C9JU48   \n",
       "31              Q9HCJ3-2                         Q9HCJ3-2;Q9HCJ3   \n",
       "32                Q9UMN6                         Q9UMN6;Q9UMN6-2   \n",
       "33                Q5SRE5                         Q5SRE5;Q5SRE5-2   \n",
       "34                Q9Y2U8                                  Q9Y2U8   \n",
       "35                Q96IV0                Q96IV0;Q96IV0-2;Q96IV0-3   \n",
       "36                Q15596                           Q15596;E7EWM1   \n",
       "37                Q12972                                  Q12972   \n",
       "38              Q96L91-5       Q96L91-5;Q96L91-2;Q96L91;Q96L91-3   \n",
       "39                P52701  P52701;P52701-3;P52701-2;F8WAX8;F8W7G9   \n",
       "40                P42566                           P42566;B1AUU8   \n",
       "41                P55036                           P55036;H0Y3Y9   \n",
       "42                Q5VWC4                                  Q5VWC4   \n",
       "43                A1X283                                  A1X283   \n",
       "44                Q8N1B4                           Q8N1B4;E9PI03   \n",
       "45                Q9BQE4                Q9BQE4;A0A182DWI4;E9PN30   \n",
       "46                Q96K37                                  Q96K37   \n",
       "47            A0A087WUC6         A0A087WUC6;Q15005;E9PL01;E9PI68   \n",
       "48                Q8N1G4                                  Q8N1G4   \n",
       "\n",
       "    Reporter_intensity_count Cell_line  Treatment State Replicate  \\\n",
       "0                          2      A549          0     D      Rep1   \n",
       "1                          1      A549          0     D      Rep1   \n",
       "2                          1      A549          0     D      Rep1   \n",
       "3                          3      A549          0     D      Rep1   \n",
       "4                          7      A549          0     D      Rep1   \n",
       "5                          1      A549          0     D      Rep1   \n",
       "6                          2      A549          0     D      Rep1   \n",
       "7                          2      A549          0     D      Rep1   \n",
       "8                          1      A549          0     D      Rep1   \n",
       "9                          1      A549          0     D      Rep1   \n",
       "10                         1      A549          0     D      Rep1   \n",
       "11                         2      A549          0     D      Rep1   \n",
       "12                         1      A549          0     D      Rep1   \n",
       "13                         1      A549          0     D      Rep1   \n",
       "14                         2      A549          0     D      Rep1   \n",
       "15                         1      A549          0     D      Rep1   \n",
       "16                         1      A549          0     D      Rep1   \n",
       "17                         1      A549          0     D      Rep1   \n",
       "18                         1      A549          0     D      Rep1   \n",
       "19                         1      A549          0     D      Rep1   \n",
       "20                         1      A549          0     D      Rep1   \n",
       "21                         1      A549          0     D      Rep1   \n",
       "22                         1      A549          0     D      Rep1   \n",
       "23                         2      A549          0     D      Rep1   \n",
       "24                         1      A549          0     D      Rep1   \n",
       "25                         1      A549          0     D      Rep1   \n",
       "26                         1      A549          0     D      Rep1   \n",
       "27                         1      A549          0     D      Rep1   \n",
       "28                         2      A549          0     D      Rep1   \n",
       "29                         3      A549          0     D      Rep1   \n",
       "30                         1      A549          0     D      Rep1   \n",
       "31                         1      A549          0     D      Rep1   \n",
       "32                         1      A549          0     D      Rep1   \n",
       "33                         1      A549          0     D      Rep1   \n",
       "34                         1      A549          0     D      Rep1   \n",
       "35                         1      A549          0     D      Rep1   \n",
       "36                         1      A549          0     D      Rep1   \n",
       "37                         2      A549          0     D      Rep1   \n",
       "38                         2      A549          0     D      Rep1   \n",
       "39                         2      A549          0     D      Rep1   \n",
       "40                         2      A549          0     D      Rep1   \n",
       "41                         2      A549          0     D      Rep1   \n",
       "42                         1      A549          0     D      Rep1   \n",
       "43                         1      A549          0     D      Rep1   \n",
       "44                         1      A549          0     D      Rep1   \n",
       "45                         2      A549          0     D      Rep1   \n",
       "46                         2      A549          0     D      Rep1   \n",
       "47                         1      A549          0     D      Rep1   \n",
       "48                         2      A549          0     D      Rep1   \n",
       "\n",
       "    Reporter_intensity_corrected  \n",
       "0                            0.0  \n",
       "1                            0.0  \n",
       "2                            0.0  \n",
       "3                            0.0  \n",
       "4                            0.0  \n",
       "5                            0.0  \n",
       "6                            0.0  \n",
       "7                            0.0  \n",
       "8                            0.0  \n",
       "9                            0.0  \n",
       "10                           0.0  \n",
       "11                           0.0  \n",
       "12                           0.0  \n",
       "13                           0.0  \n",
       "14                           0.0  \n",
       "15                           0.0  \n",
       "16                           0.0  \n",
       "17                           0.0  \n",
       "18                           0.0  \n",
       "19                           0.0  \n",
       "20                           0.0  \n",
       "21                           0.0  \n",
       "22                           0.0  \n",
       "23                           0.0  \n",
       "24                           0.0  \n",
       "25                           0.0  \n",
       "26                           0.0  \n",
       "27                           0.0  \n",
       "28                           0.0  \n",
       "29                           0.0  \n",
       "30                           0.0  \n",
       "31                           0.0  \n",
       "32                           0.0  \n",
       "33                           0.0  \n",
       "34                           0.0  \n",
       "35                           0.0  \n",
       "36                           0.0  \n",
       "37                           0.0  \n",
       "38                           0.0  \n",
       "39                           0.0  \n",
       "40                           0.0  \n",
       "41                           0.0  \n",
       "42                           0.0  \n",
       "43                           0.0  \n",
       "44                           0.0  \n",
       "45                           0.0  \n",
       "46                           0.0  \n",
       "47                           0.0  \n",
       "48                           0.0  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "melted"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "### 14:57 Protein Classification. \n",
    "Checked out reactome, but honestly I have no idea how to use this. Webscraping their uniprot2reactome.txt urls will take too much time. What did my PI mean with looking at reactome for protein classification?\n",
    "\n",
    "I got some information from my collaborative partner, A, that I should check out StringDB, CytoScape and Corum for protein classification by thier functions. \n",
    "\n",
    "Previous entry on curl on uniprot, was unnescessary, since the uniprot tab-seperated files could include GO in the columns.\n",
    "\n",
    "ToDo:\n",
    "- Check reactome more thoroughly,\n",
    "- Check StringDB, CytoScape and Corum"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2020-10-20 Tuesday\n",
    "### 12:02 - curl on Uniprot.\n",
    "\n",
    "I might be able to speed up the collection, by using bash curl and multiprocessing to to collect data from uniprot GO. This should be much quicker than python requests.\n",
    "\n",
    "This means I will have to learn:\n",
    "- multiprocessing on bash.\n",
    "- curl to get uniprot info."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2020-10-18\n",
    "I've applied for resource allocation at Swedish National Infrastucture for Computing (SNIC) and I'm waiting for the SNIC proposal to go through so I can get my gene-ontology (GO) protein classification for lifeAndDeath project in SNIC with multicore processing. Using my 8 local cores, It would theoretically take 10 days to get the GO protein classifications.\n",
    "\n",
    "Setting up a Rmarkdown log for this project. "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
